domain/application/service/unit.go

```diff
@@ -341,15 +335,49 @@ func (s *Service) GetUnitDisplayStatus(ctx context.Context, unitName coreunit.Na
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
 	containerStatus, err := s.st.GetUnitCloudContainerStatus(ctx, unitUUID)
-	if errors.Is(err, applicationerrors.UnitStatusNotFound) {
-		return unitDisplayStatus(workloadStatus, nil)
-	} else if err != nil {
+	if err != nil && !errors.Is(err, applicationerrors.UnitStatusNotFound) {
 		return nil, errors.Trace(err)
 	}
 	return unitDisplayStatus(workloadStatus, containerStatus)
 }
 
+// GetUnitAndAgentDisplayStatus returns the unit and agent display status of the
+// specified unit. The display status a function of both the unit workload status
+// and the cloud container status. It returns an error satisfying
+// [applicationerrors.UnitNotFound] if the unit doesn't exist.
+func (s *Service) GetUnitAndAgentDisplayStatus(ctx context.Context, unitName coreunit.Name) (agent *corestatus.StatusInfo, workload *corestatus.StatusInfo, _ error) {
```

Comment:
`GetUnitAgentAndDisplayStatuses` 

`AgentDisplayStatus` is not a defined entity